tests/integration-service/integration.go

```diff
@@ -46,245 +46,247 @@ var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests
+						return nil
+					}
+					return fmt.Errorf("tests haven't succeeded yet for snapshot %s/%s", snapshot_push.GetNamespace(), snapshot_push.GetName())
+				}, timeout, interval).Should(Succeed(), fmt.Sprintf("time out when waiting for updating the global candidate in %s namespace", testNamespace))
 			})
 
-			It("creates a ReleasePlan and an environment", func() {
-				_, err = f.AsKubeAdmin.ReleaseController.CreateReleasePlan(autoReleasePlan, testNamespace, applicationName, targetReleaseNamespace, "")
-				Expect(err).ShouldNot(HaveOccurred())
-				env, err = f.AsKubeAdmin.GitOpsController.CreatePocEnvironment(EnvironmentName, testNamespace)
-				Expect(err).ShouldNot(HaveOccurred())
-				testScenarios, err := f.AsKubeAdmin.IntegrationController.GetIntegrationTestScenarios(applicationName, testNamespace)
-				Expect(err).ShouldNot(HaveOccurred())
-				for _, testScenario := range *testScenarios {
-					GinkgoWriter.Printf("IntegrationTestScenario %s is found\n", testScenario.Name)
-				}
+			It("checks if a Release is created successfully", func() {
+				timeout = time.Second * 60
+				interval = time.Second * 5
+				Eventually(func() error {
+					_, err := f.AsKubeAdmin.ReleaseController.GetReleases(testNamespace)
+					return err
+				}, timeout, interval).Should(Succeed(), fmt.Sprintf("time out when waiting for release created for snapshot %s/%s", snapshot_push.GetNamespace(), snapshot_push.GetName()))
 			})
 
-			It("creates an snapshot of push event", func() {
-				sampleImage := "quay.io/redhat-appstudio/sample-image@sha256:841328df1b9f8c4087adbdcfec6cc99ac8308805dea83f6d415d6fb8d40227c1"
-				snapshot_push, err = f.AsKubeAdmin.IntegrationController.CreateSnapshotWithImage(componentName, applicationName, testNamespace, sampleImage)
-				Expect(err).ShouldNot(HaveOccurred())
+			It("checks if an SnapshotEnvironmentBinding is created successfully", func() {
+				timeout = time.Second * 600
+				interval = time.Second * 2
+				Eventually(func() error {
+					_, err := f.AsKubeAdmin.CommonController.GetSnapshotEnvironmentBinding(applicationName, testNamespace, env)
+					return err
+				}, timeout, interval).Should(Succeed(), fmt.Sprintf("timed out when waiting for SnapshotEnvironmentBinding to be created for application %s/%s", testNamespace, applicationName))
 			})
+		})
+	})
 
-			When("An snapshot of push event is created", func() {
-				It("checks if all of the integrationPipelineRuns created by push event passed", Label("slow"), func() {
-					Expect(f.AsKubeAdmin.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshot_push)).To(Succeed(), "Error when waiting for one of the integration pipelines to finish in %s namespace", testNamespace)
-				})
+	Describe("with an integration test fail", Ordered, func() {
+		BeforeAll(func() {
+			// Initialize the tests controllers
+			f, err = framework.NewFramework(IntegrationServiceUser)
```

Comment:
since this test and `with happy path for general flow of Integration service` test are both initializing framework with prefix from `IntegrationServiceUser` variable, they are both creating/deleting resources in the same namespace (if they are running in parallel). I think the best practice is to create separate namespace for each test so no conflicts can arise, wdyt?

Reply:
Thanks for catching this! Updated.